### ISSUE TITLE
OpenSSL: Remove deprecated non EVP code

### ISSUE THREAD

**Issue by matejk** [pocoproject/poco#4602]
Plenty of deprecated code is removed in #4426.

Removing non-EVP OpenSSL is a larger task that shall be handled separately.

**Comment by matejk** [comment]
## Analysis

This is the umbrella tracking issue for removing deprecated non-EVP OpenSSL code. The scope overlaps with #4846 — resolving #4846 (replacing deprecated RSA/EC_KEY types with EVP_PKEY-based API) would effectively close this issue as well.

### Current State

PR #4426 already removed some deprecated code. The remaining non-EVP code is the same set identified in #4846:
- RSA struct operations in `RSAKeyImpl`, `RSACipherImpl`, `RSADigestEngine`
- EC_KEY struct operations in `ECKeyImpl`, `ECDSADigestEngine`
- Legacy conversion functions (`EVP_PKEY_get1_RSA`, `EVP_PKEY_get1_EC_KEY`, etc.) in `EVPPKey.cpp`

### Recommendation

Close this issue together with #4846 once the EVP migration is complete. The work is identical — this issue just tracks it from a different angle (removing deprecated code vs. replacing with modern equivalents).

**Comment by matejk** [comment]
Addressed by PR #5313 which removes all non-EVP code on the OpenSSL 3.0+ path.

RSA operations (`RSA_new/free/size/generate_key_ex`, `RSA_public_encrypt/private_decrypt`, `RSA_sign/verify`, `PEM_read/write_bio_RSA*`) and EC operations (`EC_KEY_new/free/generate_key`, `ECDSA_sign/verify`, `EVP_PKEY_get1/set1_RSA/EC_KEY`) are replaced with EVP_PKEY equivalents. The non-EVP code remains in `#else` branches for OpenSSL 1.1.x only.

Duplicate of #5296.
